mtcpclient/reporting.go

```diff
@@ -38,3 +38,45 @@ func StartBackgroundReporting(numberConnections int, rinterval int) (chan tcpcli
 
 	return connStatusCh, connStatusTracker
 }
+
+func FinalMetricsReport(connectionDescriptions []tcpclient.Connection) string {
+	mr := calculateMetricsReport(connectionDescriptions)
+	return "Time to establish TCP connections min/avg/max/stdDev = " +
+		mr.minToEstablished.String() + "/" +
+		mr.avgToEstablished.String() + "/" +
+		mr.maxToEstablished.String() + "/" +
+		mr.stdDevToEstablished.String()
+}
+
+type metricsReport struct {
+	avgToEstablished    time.Duration
+	minToEstablished    time.Duration
+	maxToEstablished    time.Duration
+	totalToEstablished  time.Duration
+	stdDevToEstablished time.Duration
+}
+
+func calculateMetricsReport(c []tcpclient.Connection) metricsReport {
+	var mr metricsReport
+
+	for i, item := range c {
+		if i == 0 {
+			mr.totalToEstablished = tcpclient.TCPProcessingTime(item)
+			mr.minToEstablished = tcpclient.TCPProcessingTime(item)
+			mr.maxToEstablished = tcpclient.TCPProcessingTime(item)
+		} else {
+			mr.minToEstablished = time.Duration(math.Min(float64(mr.minToEstablished), float64(tcpclient.TCPProcessingTime(item))))
+			mr.maxToEstablished = time.Duration(math.Max(float64(mr.maxToEstablished), float64(tcpclient.TCPProcessingTime(item))))
+			mr.totalToEstablished += tcpclient.TCPProcessingTime(item)
+		}
+	}
+	mr.avgToEstablished = mr.totalToEstablished / time.Duration(len(c))
+
+	var sd float64
```

Comment:
what about of moving the std deviation calculation to a single function?

Reply:
👍